tracer.go

```diff
@@ -0,0 +1,290 @@
+package pubsub
```

Comment:
Double checking, this is moving to a new repo?

Reply:
I think we can keep this file here, so that we have a one-stop construction of the pubsub system without requiring an external dependency.